tests/python/test_python_frontend.py

```diff
@@ -4016,6 +4016,187 @@ def fusion_func(fd: FusionDefinition) -> None:
 
         nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
 
+    # See https://github.com/NVIDIA/Fuser/issues/2317
+    @unittest.skipIf(is_pre_ampere(), "Only supported on Ampere and newer devices.")
+    def test_reduction_transpose_sched_issue2317(self):
```

Comment:
Is it possible to create a smaller, more targeted unit test? I understand that the test as is reproduces the issue, but given the size it'll be hard to debug when it fails.

Reply:
Let me give it a shot.

Reply:
I agree. Ideally we would just have a fusion with a sum along some axes, and a transpose. It would be nice to exercise multiple reduction axes as well given that the reduction skipping code is slightly tricky.

Reply:
I added a smaller repro, using `linear`. I ran into another issue when using sum:

```python
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id0(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1, -1, -1, -1, -1], contiguity=[True, True, True, True, True], dtype=DataType.BFloat16, is_cpu=False, stride_order=[4, 3, 2, 1, 0])
    T1 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.BFloat16, is_cpu=False, stride_order=[3, 2, 1, 0])
    T2 = fd.ops.permute(T0, dims=[0, 1, 3, 2, 4])
    T3 = fd.ops.stride_order(T2, stride_order=[4, 3, 2, 1, 0])
    V4 = fd.ops.shape(T1)
    T5 = fd.ops.reshape(T3, new_shape=V4)
    T6 = fd.ops.sum(T5, dims=[1], keepdim=False, dtype=DataType.Null)
    T7 = fd.ops.sum(T1, dims=[1], keepdim=False, dtype=DataType.Null)
    T8 = fd.ops.add(T6, T7)
    fd.add_output(T8)

with FusionDefinition() as fd:
    nvfuser_fusion_id0(fd)

inputs = [
    torch.randn((104857600,), dtype=torch.bfloat16, device='cuda:0').as_strided((16, 32, 25, 128, 64), (6553600, 204800, 8192, 64, 1)),
    torch.randn((104857600,), dtype=torch.bfloat16, device='cuda:0').as_strided((16, 32, 128, 1600), (6553600, 204800, 1600, 1)),
]
fd.execute(inputs)

Traceback (most recent call last):
  File "/opt/pytorch/nvfuser/nvfuser/__init__.py", line 139, in execute
    result = self._execute(
RuntimeError: false INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/executor_utils.cpp":829, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. 
```

Reply:
Ahh the repro is not correct. Let me fix it.

Reply:
It looks like we're not promoting to float in that sum case. Would you mind posting that as a separate issue?